src/impl/realm_coordinator.cpp

```diff
@@ -53,16 +59,46 @@ std::shared_ptr<RealmCoordinator> RealmCoordinator::get_coordinator(StringData p
     return coordinator;
 }
 
+std::shared_ptr<RealmCoordinator> RealmCoordinator::get_coordinator(const Realm::Config& config)
+{
+    auto coordinator = get_coordinator(config.path);
+    std::lock_guard<std::mutex> lock(coordinator->m_realm_mutex);
+    coordinator->set_config(config);
+    return coordinator;
+}
+
 std::shared_ptr<RealmCoordinator> RealmCoordinator::get_existing_coordinator(StringData path)
 {
     std::lock_guard<std::mutex> lock(s_coordinator_mutex);
     auto it = s_coordinators_per_path.find(path);
     return it == s_coordinators_per_path.end() ? nullptr : it->second.lock();
 }
 
-std::shared_ptr<Realm> RealmCoordinator::get_realm(Realm::Config config)
+void RealmCoordinator::create_sync_session()
+{
+#if REALM_ENABLE_SYNC
+    if (m_sync_session)
+        return;
+
+    m_sync_session = SyncManager::shared().get_session(m_config.path, *m_config.sync_config);
+
+    std::weak_ptr<RealmCoordinator> weak_self = shared_from_this();
+    SyncSession::Internal::set_sync_transact_callback(*m_sync_session,
+                                                      [weak_self](VersionID old_version, VersionID new_version) {
+        if (auto self = weak_self.lock()) {
+            if (self->m_transaction_callback)
+                self->m_transaction_callback(old_version, new_version);
+            self->notify_others();
+        }
+    });
+    if (m_config.sync_config->error_handler) {
+        SyncSession::Internal::set_error_handler(*m_sync_session, m_config.sync_config->error_handler);
```

Comment:
Yeah, that looks good. Are there any changes I need to make to the API to accommodate your use case?